src/Service/ManifestCreationService.php

```diff
@@ -0,0 +1,114 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\AppBundle\Service;
+
+use Shopware\AppBundle\Attribute\Webhook;
+use Shopware\AppBundle\Metadata;
+use SimpleXMLElement;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class ManifestCreationService
```

Comment:
Also here we don't have to tell others that it is a service :D We can name it `ManifestCreator` or `ManifestGenerator`

Reply:
IMO the best way to implement the generator is to have a composition of many smaller services each creating a specific XML element. Think about something like this.

```
public function generate(): string
{
    // first create empty document and root element
    $document = new DOMDocument('1.0', 'utf-8');
    $document->formatOutput = true;  // TE must read this when it is published :D

    $manifest = $document->createElement('manifest');
    $document->appendChild($manifest)

    $manifest->appendChild($this->metaDataGenerator->generate());

    $setup = $this->setupGenerator->generate();

    if ($setup !== null) {
        $manifest->appendChild($setup);
    }
    
    // add other childelements...

   // return formated XML;
   return $document->saveXML();
}
```